how-it-works/1_protocol/01_peer-to-peer-p2p.card.md

```diff
@@ -4,8 +4,14 @@ title: Peer-to-peer layer
 
 ![](/img/how-it-works/peer-to-peer-p2p.600x300.jpg)
 
-# Peer-to-peer layer
+# Peer-to-peer
 
-The Internet Computer blockchain is designed to be secure, reliable, and scalable. Scalability is provided through subnet blockchains that run Canister smart contracts on a selection of nodes. The peer-to-peer layer ("P2P") enables reliable and secure communication between nodes participating in the same subnet. P2P collects and advertises messages from users, as well as from other nodes, so that they will eventually be replicated across all nodes in the same subnet blockchain.
+The peer-to-peer layer (P2P) of the Internet Computer, the bottommost layer in the protocol stack, is responsible for the secure and reliable communication between the nodes of a subnet.
+The P2P layer realizes a virtual peer-to-peer broadcast network between the nodes of a subnet, building upon the Internet Protocol (IP) connectivity between the nodes.
+This makes the P2P layer the communications fabric that connects all the nodes of a subnet.
+Using P2P, a node can broadcast a *message*, also called *artifact*, to all the nodes in the subnet.
+Artifacts can be things like ingress messages submitted by users or protocol messages (e.g., signature shares) generated by the IC protocol.
```

Comment:
Really, "signature shares"? This is completely out of context. Better example? idk...signature on a block?

Reply:
I would use "block proposals" here